onnxruntime/core/optimizer/graph_transformer_utils.cc

```diff
@@ -146,14 +146,21 @@ std::vector<std::unique_ptr<GraphTransformer>> GenerateTransformers(
     const IExecutionProvider& cpu_execution_provider, /*required by constant folding*/
     const std::unordered_set<std::string>& rules_and_transformers_to_disable) {
   std::vector<std::unique_ptr<GraphTransformer>> transformers;
-  std::unique_ptr<RuleBasedGraphTransformer> rule_transformer = nullptr;
-  bool disable_quant_qdq = session_options.config_options.GetConfigOrDefault(kOrtSessionOptionsDisableQuantQDQ, "0") == "1";
+  bool disable_quant_qdq =
+      session_options.config_options.GetConfigOrDefault(kOrtSessionOptionsDisableQuantQDQ, "0") == "1";
 #ifndef DISABLE_CONTRIB_OPS
-  bool enable_gelu_approximation = session_options.config_options.GetConfigOrDefault(kOrtSessionOptionsEnableGeluApproximation, "0") == "1";
+  bool enable_gelu_approximation =
+      session_options.config_options.GetConfigOrDefault(kOrtSessionOptionsEnableGeluApproximation, "0") == "1";
 #endif
 
   switch (level) {
     case TransformerLevel::Level1: {
+      // RewriteRule transformers are the simplest so run them first
```

Comment:
>RewriteRule transformers are the simplest so run them first [](http://example.com/codeflow?start=9&length=59)

perhaps I'm missing something, but why is being simplest a good reason to run them first?

Reply:
they tend to remove unnecessary nodes and are lower cost to run, so reduces the work for more complicated optimizers like constant folding (which actually executes nodes so is more expensive).

Reply:
Updated comment to add this detail